sdk/storage/blobs/src/az_storage_blobs_blob_client.c

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_json_builder.h>
+#include <az_storage_blobs.h>
+
+#include <_az_cfg.h>
+
+/**
+ * @brief Maximum allowed URL size:
+ * url is expected as : [https://]{account_id}[.blob.core.windows.net]/{container}/{blob}{optional_query}
+ * URL token                       max Len            Total
+ * [https://]                       = 8                 8
+ * {account_id}                     = 24               32  // 3-24 lowercase only
+ * [.blob.core.windows.net]         = 22               54  //TODO: Support soveriegn clouds
+ * /{container}                     = 63              117  // 3-63, lowercase letters/numbers/- (consecutive dash is invalid)
+ * /{blob}                          = 1024           1141
+ * {optional_query}                 = 70          ** 1211 **
+ */
+enum { MAX_URL_SIZE = 1211 };
+enum { MAX_BODY_SIZE = 1024 * 10 }; // 10KB buffer  /*TODO, Adjust this to reasonable size for non-stream data.*/
+
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_client_constant_date() { return AZ_STR("x-ms-date");}
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_client_constant_blob_type() { return AZ_STR("x-ms-blob-type");}
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_client_constant_content_length() { return AZ_STR("Content-Length");}
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_client_constant_content_type() { return AZ_STR("Content-Type");}
+AZ_NODISCARD AZ_INLINE az_span az_storage_blobs_client_constant_text_plain() { return AZ_STR("text/plain; charset=UTF-8");}
+
+az_storage_blobs_blob_client_options const AZ_STORAGE_BLOBS_BLOB_CLIENT_DEFAULT_OPTIONS
+    = { .service_version = AZ_CONST_STR("2015-02-21"),
```

Comment:
We'll need to have a conversation about how to best handle service API versions.